pennylane/ops/qubit/observables.py

```diff
@@ -435,14 +435,22 @@ class BasisStateProjector(Projector, Operation):
     r"""Observable corresponding to the state projector :math:`P=\ket{\phi}\bra{\phi}`, where
     :math:`\phi` denotes a basis state."""
 
+    grad_method = None
+
     # The call signature should be the same as Projector.__new__ for the positional
     # arguments, but with free key word arguments.
     def __init__(self, state, wires, id=None):
         wires = Wires(wires)
-        state = list(qml.math.toarray(state).astype(int))
 
-        if not set(state).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {state}")
+        if qml.math.get_interface(state) == "jax":
+            if qml.math.dtype(state).name[:3] not in {"int", "boo"}:
```

Comment:
Would `np.issubdtype(dtype, np.integer) or np.issubdtype(dtype, bool_)` work here, since jax comes from numpy? Maybe that's not better than a string comparison anyway.

Reply:
Seems to work. I'll give that a try.